### ISSUE TITLE
Unknown logger errors resulting in exiting without an error message

### ISSUE THREAD

**Issue by NavyAdmiral** [smartcontractkit/chainlink#835]
Hello. This was pretty frustrating. `chainlink n` exited without any message whatsoever:

<p align="center">
<img src="https://i.imgur.com/Aw5EbuP.png"/>
</p>

I had time to debug it step by step and see where it was failing and found out that it was due to Logger unable to write to disk. Ultimately the failure was observed in:
```go
// CreateProductionLogger returns a log config for the passed directory
// with the given LogLevel and customizes stdout for pretty printing.
func CreateProductionLogger(
	dir string, jsonConsole bool, lvl zapcore.Level, toDisk bool) *zap.Logger {
	config := zap.NewProductionConfig()
	if !jsonConsole {
		config.OutputPaths = []string{"pretty://console"}
	}
	if toDisk {
		destination := ProductionLoggerFilepath(dir)
		config.OutputPaths = append(config.OutputPaths, destination)
		config.ErrorOutputPaths = append(config.ErrorOutputPaths, destination)
	}
	config.Level.SetLevel(lvl)

	zl, err := config.Build(zap.AddCallerSkip(1), zap.AddStacktrace(zapcore.WarnLevel))
	if err != nil {
		log.Fatal(err) // NAVYADMIRAL - EXITS HERE
	}
	return zl
}
```

Now I also stepped in here:
```go
	if err != nil {
		log.Fatal(err) // NAVYADMIRAL - EXITS HERE
	}
```

and saw that Fatal is implemented in the following way:

```go
func Fatal(v ...interface{}) {
	std.Output(2, fmt.Sprint(v...))
	os.Exit(1)
}
```

I think the way std.Output is implemented is lacking in something because in the debugger I was able to see the error message:
```

v[0].data.s = "couldn't open sink "C:\\Users\\Users\\.chainlink/log.jsonl"...+30 more
```

However without debugger I wasn't able to see the error message anywhere. For clarity, here's the msg in debugger:

![image](https://user-images.githubusercontent.com/40662484/49844825-623cbb80-fd92-11e8-8635-2b39544cf280.png)
I suppose `fmt.Sprint(v...)` is responsible for displaying it.

I was able to fix this issue by setting `LOG_TO_DISK false`. Even now if I switch back to log to disk true, chainlink will exit without an error message leaving it up to you to figure out what's going on.

Ultimately, I didn't figure out why `couldn't open sink "C:\\Users\\Users\\.chainlink/log.jsonl` happened. Since this happened on Windows 10, could it be https://github.com/uber-go/zap/issues/621 ?

Even if this is resolved by fixing that zap issue, I think there's something wrong with log Fatal() that needs to be addressed.

**Comment by whatsadebugger** [comment]
Is C:\Users\Users\.chainlink a valid path on your machine?

**Comment by NavyAdmiral** [comment]
> Is C:\Users\Users.chainlink a valid path on your machine?

yes

**Comment by NavyAdmiral** [comment]
Addressed by #977
